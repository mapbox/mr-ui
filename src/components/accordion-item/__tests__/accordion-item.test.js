import React from 'react';
import { shallow, mount } from 'enzyme';
import { testCases } from './accordion-item-test-cases';

describe('Accordion', () => {
  let testCase;
  let wrapper;

  describe(testCases.basic.description, () => {
    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = mount(React.createElement(testCase.component, testCase.props));
    });

    test('renders', () => {
      expect(
        shallow(React.createElement(testCase.component, testCase.props))
      ).toMatchSnapshot();
    });

    test('internal onToggle is called', () => {
      wrapper.find('button').first().simulate('click');
      expect(testCase.props.onToggle).toHaveBeenCalledTimes(1);
      expect(testCase.props.onToggle).toHaveBeenCalledWith('one');
    });
  });

  describe(testCases.disabled.description, () => {
    beforeEach(() => {
      testCase = testCases.disabled;
      wrapper = mount(React.createElement(testCase.component, testCase.props));
    });

    test('renders', () => {
      expect(
        shallow(React.createElement(testCase.component, testCase.props))
      ).toMatchSnapshot();
    });

    test('disabled works', () => {
      const props = wrapper.find('button').last().props();
      expect(props).toHaveProperty('disabled', true);
    });
  });

  describe(testCases.allOptions.description, () => {
    beforeEach(() => {
      testCase = testCases.allOptions;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('extra props works', () => {
      expect(testCase.props).toHaveProperty('onToggle');

      // Optional props passed to items
      expect(testCase.props).toHaveProperty('active', true);
      expect(testCase.props).toHaveProperty('themeItem');
      expect(testCase.props).toHaveProperty('themeItemHeader');
      expect(testCase.props).toHaveProperty('themeItemBody');
    });
  });
});
