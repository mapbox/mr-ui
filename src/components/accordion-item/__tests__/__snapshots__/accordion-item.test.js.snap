// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion all options renders 1`] = `
<div
  className="bg-red-faint border-b border--red-dark"
>
  <button
    aria-label="Toggle"
    className="w-full bg-red txt-l txt-bold link link--white bg-red-dark-on-hover px6 py6 cursor-pointer"
    data-test="one"
    onClick={[Function]}
  >
    <span>
      one
    </span>
  </button>
  <div
    className="py6 px6 txt-l bg-red-light"
  >
    First body
  </div>
</div>
`;

exports[`Accordion basic renders 1`] = `
<div>
  <button
    aria-label="Toggle"
    className="w-full txt-s txt-bold txt-truncate link link--gray py6 cursor-pointer"
    data-test="one"
    onClick={[Function]}
  >
    <span>
      one
    </span>
  </button>
</div>
`;

exports[`Accordion disabled renders 1`] = `
<div>
  <button
    aria-label="Toggle"
    className="w-full txt-s txt-bold txt-truncate link link--gray py6 cursor-default color-gray-light"
    data-test="one"
    disabled={true}
    onClick={[Function]}
  >
    <span>
      one
    </span>
  </button>
</div>
`;
